cmd/openshift-install/gather.go

```diff
@@ -114,25 +114,26 @@ func runGatherBootstrapCmd(directory string) (string, error) {
 
 		for _, stage := range platformstages.StagesForPlatform(config.Config.Platform.Name()) {
 			stageBootstrap, stagePort, stageMasters, err := stage.ExtractHostAddresses(directory, config.Config)
-			if err != nil {
-				return "", err
-			}
-			if stageBootstrap != "" {
-				bootstrap = stageBootstrap
+			if err != nil || stageBootstrap == "" {
+				return "", errors.Wrapf(err, "failed to obtain IP address for bootstrap host")
+			} else if err != nil {
+				logrus.Debugf("Error when obtaining host IP addresses, continuing with best effort. Error message: %s", err)
 			}
```

Comment:
This logic is not quite right. (1) It should be done *after* going through all of the stages. (2) An error in extracting the host addresses does not imply that there was an error obtaining the IP address specifically for the bootstrap host. Keep in mind that each stage may have all, some, or none of the host addresses, depending upon the platform.